aiida/tools/importexport/dbimport/backends/common.py

```diff
@@ -66,7 +66,7 @@ def collect_hashkeys(objects, hashkeys):
         container_export.export(set(hashkeys), container_profile, compress=True, callback=callback)
 
 
-def _make_import_group(*, group: Optional[ImportGroup], node_pks: List[int]) -> ImportGroup:
+def _make_import_group(*, group: Optional[ImportGroup], node_pks: List[int]) -> Optional[ImportGroup]:
```

Comment:
To fix the mypy error related to inconsistent types, we should rather change this to `Optional[Group]`. I have already done this and reverted your other change. Reason why is explained in another comment